File: code_samples/scala/web_development_middleware.scala
/**
 * Web Development: Middleware
 * AI/ML Training Sample
 */

class Middleware {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object MiddlewareExample {
  def main(args: Array[String]): Unit = {
    val instance = new Middleware()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}

File: code_samples/haskell/web_development_middleware.hs
-- Web Development: Middleware
-- AI/ML Training Sample

module Middleware where

data Middleware = Middleware {
    getData :: String
} deriving (Show, Eq)

process :: Middleware -> String -> Middleware
process obj input = obj { getData = input }

validate :: Middleware -> Bool
validate obj = not (null (getData obj))

main :: IO ()
main = do
    let instance = Middleware { getData = "" }
    let updated = process instance "example"
    putStrLn $ "Data: " ++ getData updated
    putStrLn $ "Valid: " ++ show (validate updated)
